dataclasses_avroschema/fields/field_utils.py

```diff
@@ -97,3 +98,11 @@
     datetime.datetime: {"type": LONG, "logicalType": TIMESTAMP_MILLIS},
     uuid.uuid4: {"type": STRING, "logicalType": UUID},
 }
+
+
+def ensure_null_first(avro_type: typing.Any) -> typing.Any:
```

Comment:
This is not needed. Now in `avro` default values do not have to match the first `union schema`. You can check [this](https://github.com/apache/avro/pull/2503). fastavro has been updated as well.

Reply:
[Avrotool](https://github.com/exaspace/avrotool) was crashing when "null" wasn't first.  Since that code is based on  the Confluent Schema validator, the format may not be wrong but it seems like it would be problem in practice.

Reply:
It also likely means that Confluent's schema registry code must be updated.  Are you sure it is worth being a purist about this, if the consequence is incompatibility with a substantial deployed Avro infrastructure?

Reply:
Hard to tell. I prefer to move forward but probably a lot of people are using this library with `Confluent's schema registry`. For the rest of the clients I think it won't break anything as the change in  avro was intriduced no so long ago, but it might create some confusion. We can keep you solution and keep track of `Confluent's schema registry`. If other clients are complaining then we should remove the check.

Reply:
As far as I can tell, moving "null" to the front doesn't break anything.  So hopefully it can remain until it is no longer needed.

I've asked our Confluent tech support to let us know if this is a known issue in their validator if/when it was/will be fixed.  I'll directly test in our Kafka staging environment when an internal product release verification is complete.

Reply:
I don't have an ETA on when Confluent will get back to me.  How about I strip this change from this PR and we can revisit once we know if contemporary Confluent deployments still suffer from this bug?

Reply:
yes, do it. Let's split your work in two. Just finish the new feat `Pydantic field alias and serialization_alias`. Thanks!